doc/Compilation.md

```diff
@@ -178,6 +178,39 @@ sections are placed in the expected order, adding a section that lists
 necessary load-time relocations, and creating the `LoadInfo` header.
 
 
+### Tock Application Bundle
+
+To support ease-of-use and distributable applications, Tock applications are
+compiled for multiple architectures and bundled together into a "Tock
+Application Bundle" or `.tab` file. This creates a standalone file for an
+application that can be flashed onto any board that supports Tock, and removes
+the need for the board to be specified when the application is compiled.
+The TAB has enough information to be flashed on many or all Tock compatible
+boards, and the correct binary is chosen when the application is flashed
+and not when it is compiled.
+
+#### TAB Format
+
+`.tab` files are `tar`ed archives of TBF compatible binaries along with a
+`metadata.toml` file that includes some extra information about the application.
+A simplified example command that creates a `.tab` file is:
+
+    tar cf app.tab cortex-m0.bin cortex-m4.bin metadata.toml
+
+#### Metadata
+
+The `metadata` file in the `.tab` file is a TOML file that contains a series of
+key-value pairs, one per line, that provides more detailed information and can
+help when flashing the application. Existing fields:
+
+```
+tab-version = 1                         // Version number for this TAB
+name = "<package name>"                 // Package name of the application
+compatible-with = <list of boards>      // List of board kernels that this application supports
```

Comment:
1. I want to be careful about "or choose that board from multiple plugged in boards". That can very quickly get to a point where you have to reset boards to figure out what they are, which could make for a very awful user experience. It's no longer on the short-term roadmap.

2. And yeah, this hint would be completely optional. My sense is that it will be obvious when to use it (like in the signpost module app case or the `tests/hail` app case).